src/Instrumentation/PDO/src/PDOInstrumentation.php

```diff
@@ -122,6 +128,32 @@ public static function register(): void
                 $span->setAttributes($attributes);
 
                 Context::storage()->attach($span->storeInContext($parent));
+                if (ContextPropagation::isEnabled() && $sqlStatement !== self::UNDEFINED) {
+                    if (array_key_exists(TraceAttributes::DB_SYSTEM_NAME, $attributes)) {
+                        /** @psalm-suppress PossiblyInvalidCast */
+                        switch ((string) $attributes[TraceAttributes::DB_SYSTEM_NAME]) {
+                            case 'postgresql':
+                            case 'mysql':
+                                $comments = [];
+                                Globals::propagator()->inject($comments);
```

Comment:
@jerrytfleung the [semconv](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md#sql-commenter) states:

> The instrumentation SHOULD allow users to pass a propagator to overwrite the global propagator. If no propagator is provided by the user, instrumentation SHOULD use the global propagator.

It doesn't seem we currently support this. And side curiosity why not have the `SqlCommentPropagator` use global propagator directly instead of passing around  `$comments`?

Reply:
It is because SqlCommentPropagator is not registered in Registry. The Globals::propagator() needs environment variable OTEL_PROPAGATORS https://github.com/open-telemetry/opentelemetry-php/blob/main/src/SDK/Propagation/PropagatorFactory.php#L21 to initialize the propagator list.